### PR TITLE
Refactored code to set ErrorInfo in Info Msg tin a routine encodeErrorInfo

### PR DESCRIPTION
**Pull request by kalyan-nidumolu**
Part of splitting up the PortErr diffs..

Signed-off-by: Kalyan Nidumolu <kalyan@zadeda.com>

### CODE REVIEW

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 1]
```diff
@@ -66,6 +66,22 @@ var appPersistPaths = []string{
...
+	if errTime.IsZero() {
+		// No Success / Error to report
+		return nil
+	}
+	// Timestamp set. Even if the error is "" - the timestamp indicates the last
+	//  testing time. Worth sending it.
+	errInfo := new(info.ErrorInfo)
+	errInfo.Description = errStr
+	protoTime, err := ptypes.TimestampProto(errTime)
+	if err != nil {
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handlemetrics.go:79]: This begs the question (avoided in the original code) whether we should log.Error if we get an error back.

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 2]
```diff
@@ -66,6 +66,22 @@ var appPersistPaths = []string{
 	types.AppImgDirname,
 }
 
+func encodeErrorInfo(errStr string, errTime time.Time) *info.ErrorInfo {
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handlemetrics.go:69]: You might want to consider passing ErrorAndTime here, once the other PR makes it in.

Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/handlemetrics.go:69]: Yeah.. Need to be updated once u r PR merges up..

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 1 above)
Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/handlemetrics.go:79]: Good point.. Will add it.

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 2 above)
Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: Done..

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 1 above)
Comment (kalyan-nidumolu) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: Done.

**Pull request merged**